src/mono/mono/metadata/marshal.c

```diff
@@ -3940,10 +3960,55 @@ emit_managed_wrapper_noilgen (MonoMethodBuilder *mb, MonoMethodSignature *invoke
 }
 #endif
 
+static gboolean
+type_is_blittable (MonoType *type)
+{
+	if (type->byref)
```

Comment:
Maybe also check for `mono_type_is_reference (type)`?

Reply:
I think I want to look at the `MonoClass:blittable` bit for all of those.  This is just to get the `int&` sort of types.